### PR TITLE
serverless-agent: expose priorities control

### PR DESCRIPTION
**Pull request by gnosek**

### CODE REVIEW

**Comment by gnosek** [comment]
The linter failure is not mine

File: sysdig/data_source_sysdig_fargate_workload_agent.go [hunk 1]
```diff
@@ -429,6 +437,11 @@ func newPatchOptions(d *schema.ResourceData) *patchOptions {
 }
 
 func dataSourceSysdigFargateWorkloadAgentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	priority := d.Get("priority").(string)
+	if priority != "security" && priority != "availability" {
+		return diag.Errorf("Invalid priority: %s. must be either \"security\" or \"avaiability\"", priority)
```
Comment (pgcrooks-sysdig) [inline sysdig/data_source_sysdig_fargate_workload_agent.go:442]:
```suggestion
		return diag.Errorf("Invalid priority: %s. must be either \"security\" or \"availability\"", priority)
```

🙃

File: sysdig/data_source_sysdig_fargate_ECS_test.go [hunk 2]
```diff
@@ -164,13 +171,21 @@ func TestNewPatchOptions(t *testing.T) {
...
-		Essential: true,
+		Essential: false,
 	}
 	actualPatchOptions := newPatchOptions(data)
 
 	if !reflect.DeepEqual(expectedPatchOptions, actualPatchOptions) {
 		t.Errorf("patcConfigurations are not equal. Expected: %v, Actual: %v", expectedPatchOptions, actualPatchOptions)
 	}
+
+	data.Set("priority", "security")
```
Comment (pgcrooks-sysdig) [inline sysdig/data_source_sysdig_fargate_ECS_test.go:182]: Sadly this error is ours - there are some copy/pasta opportunities.

**Review by pgcrooks-sysdig — CHANGES_REQUESTED** [review]
Looks good apart from 2 minors

**Review by francesco-racciatti — APPROVED** [review]
lgtm apart from the minors Paul highlighted 🙂

File: sysdig/data_source_sysdig_fargate_ECS_test.go (hunk 2 above)
Comment (gnosek) [reply sysdig/data_source_sysdig_fargate_ECS_test.go:182]: oh. good catch. I haven't seen it complain about this line, must have somehow missed it

File: sysdig/data_source_sysdig_fargate_workload_agent.go (hunk 1 above)
Comment (pgcrooks-sysdig) [reply sysdig/data_source_sysdig_fargate_workload_agent.go]: It was equally hard to spot ;-)

**Review by pgcrooks-sysdig — APPROVED** [review]
LGTM

**Pull request merged**